### ISSUE TITLE
lld-link throws warnings about PDB indexes when using with latest Visual Studio 2022 (17.1.0)

### ISSUE THREAD

**Issue by tru** [llvm/llvm-project#54500]
This is split out from #54487 since they seem to be fairly unrelated at this point.

With lld-link and the latest visual studio and a command like this:
```
lld-link.exe "main.obj" /NOLOGO /DEBUG /SUBSYSTEM:CONSOLE /MACHINE:X64 /LIBPATH:"%VS2022_DIR%\VC\Tools\MSVC\14.31.31103\lib\x64" /LIBPATH:"%WINSDK_DIR%\ucrt\x64" /LIBPATH:"%WINSDK_DIR%\um\x64"
```

You get warnings like this:

```
lld-link: warning: procedure symbol record for `_FilterSetCurrentException` in d:\a01\_work\43\s\Intermediate\vctools\libvcruntime.nativeproj_520857879\objr\amd64\ehhelpers.obj refers to PDB item index 0x1191 which is not a valid function ID record
```

This warning comes from this source line: https://github.com/llvm/llvm-project/blob/main/lld/COFF/PDB.cpp#L385

I dumped the info from the [PDB](https://gist.github.com/tru/721bdb645518df2bdcd8d5f9391c60f8) and the [OBJ](https://gist.github.com/tru/dad22d679bec9cf1ed98e29c8c7e2d48) in gists.

I am trying to follow along in the code there - but unsure how the functtype is used and where it's looked up. Any pointers would be nice.

cc @aganea @sylvain-audi @rnk

**Comment by rnk** [comment]
The function symbol records (`S_[GL]PROC32_ID`) are supposed to refer to function ID records (`LF_[M]FUNC_ID`) in the type or item stream. I've forgotten what purpose this indirection serves, but the linker is responsible for translating from the `*PROC32_ID` record family to the `*PROC32` record family, which is equivalent, but which refers to the actual function type `LF_PROCEDURE`/`LF_FUNCTION` instead of the ID record.

This warning could easily be a consequence of issue #54487. If the linker loads and looks in the wrong PDB (because two PDBs have the same GUID), it's natural that item index 0x1191 might not actually be a `LF_[M]FUNC_ID` record.

The alternative explanation is that there is a new `LF_*_ID` type that LLD needs to handle. Either way, `llvm-pdbutil dump -types -ids vcruntime.pdb` will produce more readable output than the YAML dumping tool.

**Comment by tru** [comment]
Thanks @rnk for your description of the problem! 
I confirmed that this is because of the root issue in #54487 - so I'll close this one and we can discuss the fix in the other issue instead.
